server/src/main/java/org/elasticsearch/index/codec/vectors/DefaultIVFVectorsWriter.java

```diff
@@ -120,10 +100,55 @@ long[] buildAndWritePostingsLists(
             // keeping them in the same file indicates we pull the entire file into cache
             docIdsWriter.writeDocIds(j -> floatVectorValues.ordToDoc(cluster.get(j)), cluster.size(), postingsOutput);
             writePostingList(cluster, postingsOutput, binarizedByteVectorValues);
+
+            if (infoStream.isEnabled(IVF_VECTOR_COMPONENT)) {
+                clustersForMetrics[c] = cluster.toArray();
+            }
+        }
+
+        if (infoStream.isEnabled(IVF_VECTOR_COMPONENT)) {
+            printClusterQualityStatistics(clustersForMetrics, infoStream);
```

Comment:
we should be able to just do this from the IntArrayList[] objects, lets remove this `clustersForMetrics` thing, seems like its unnecessary.